### PR TITLE
Api: update command domainId/account descriptions

### PR DESCRIPTION
**Pull request by soreana**
### Description

This PR fixes the issue #7776 and the spelling issues mentioned in: https://github.com/apache/cloudstack-go/pull/64

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [x] Trivial

### How Has This Been Tested?

This PR doesn't need specific testing, it only changes some api call descriptions.

### CODE REVIEW

**Comment by soreana** [comment]
@blueorangutan package

File: api/src/main/java/org/apache/cloudstack/api/command/user/volume/CreateVolumeCmd.java [hunk 1]
```diff
@@ -70,7 +70,8 @@ public class CreateVolumeCmd extends BaseAsyncCreateCustomIdCmd implements UserC
                type = CommandType.UUID,
                entityType = DomainResponse.class,
                description = "the domain ID associated with the disk offering. If used with the account parameter"
-                   + " returns the disk volume associated with the account for the specified domain.")
+                   + " returns the disk volume associated with the account for the specified domain." +
+                       "If account is NOT provided then virtual machine will be assigned to the caller account and domain.")
```
Comment (weizhouapache) [inline api/src/main/java/org/apache/cloudstack/api/command/user/volume/CreateVolumeCmd.java:74]: volume , not virtual machine

File: api/src/main/java/org/apache/cloudstack/api/command/user/volume/UploadVolumeCmd.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public class UploadVolumeCmd extends BaseAsyncCmd implements UserCmd {
     @Parameter(name = ApiConstants.DOMAIN_ID,
                type = CommandType.UUID,
                entityType = DomainResponse.class,
-               description = "an optional domainId. If the account parameter is used, domainId must also be used.")
+               description = "an optional domainId. If the account parameter is used, domainId must also be used. If account is NOT provided then virtual machine will be assigned to the caller account and domain.")
```
Comment (weizhouapache) [inline api/src/main/java/org/apache/cloudstack/api/command/user/volume/UploadVolumeCmd.java:81]: volume , not virtual machine

File: api/src/main/java/org/apache/cloudstack/api/command/user/snapshot/CreateSnapshotCmd.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ public class CreateSnapshotCmd extends BaseAsyncCreateCmd {
     @Parameter(name = ApiConstants.DOMAIN_ID,
                type = CommandType.UUID,
                entityType = DomainResponse.class,
-            description = "The domain ID of the snapshot. If used with the account parameter, specifies a domain for the account associated with the disk volume.")
+            description = "The domain ID of the snapshot. If used with the account parameter, specifies a domain for the account associated with the disk volume. If account is NOT provided then virtual machine will be assigned to the caller account and domain.")
```
Comment (weizhouapache) [inline api/src/main/java/org/apache/cloudstack/api/command/user/snapshot/CreateSnapshotCmd.java:65]: snapshot

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java [hunk 4]
```diff
@@ -118,7 +118,8 @@ public class CreateNetworkCmd extends BaseCmd implements UserCmd {
     private Long projectId;
 
     @Parameter(name = ApiConstants.DOMAIN_ID, type = CommandType.UUID, entityType = DomainResponse.class, description = "domain ID of the account owning a network. " +
-            "If no account is provided then network will be assigned to the caller account and domain")
+            "If the account or acltype is not specified, the network will be automatically assigned to the caller account and domain." +
+            "To create a network under the domain without linking it to any account, make sure to include acltype=Domain parameter in your api call.")
```
Comment (weizhouapache) [inline api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:122]: your -> the ?

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java [hunk 5]
```diff
@@ -118,7 +118,8 @@ public class CreateNetworkCmd extends BaseCmd implements UserCmd {
     private Long projectId;
 
     @Parameter(name = ApiConstants.DOMAIN_ID, type = CommandType.UUID, entityType = DomainResponse.class, description = "domain ID of the account owning a network. " +
-            "If no account is provided then network will be assigned to the caller account and domain")
+            "If the account or acltype is not specified, the network will be automatically assigned to the caller account and domain." +
```
Comment (weizhouapache) [inline api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:121]: if account is not specified, but acltype is Domain, the network will be created for the specified domain.

**Comment by soreana** [comment]
Thanks @weizhouapache I've updated the pr, let me know if it requires more changes.

File: api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java [hunk 6]
```diff
@@ -118,7 +118,9 @@ public class CreateNetworkCmd extends BaseCmd implements UserCmd {
     private Long projectId;
 
     @Parameter(name = ApiConstants.DOMAIN_ID, type = CommandType.UUID, entityType = DomainResponse.class, description = "domain ID of the account owning a network. " +
-            "If no account is provided then network will be assigned to the caller account and domain")
+            "If the account or acltype is not specified, the network will be automatically assigned to the caller account and domain. " +
```
Comment (weizhouapache) [inline api/src/main/java/org/apache/cloudstack/api/command/user/network/CreateNetworkCmd.java:121]: line 121 could be more clear like
```
            "If the account is not specified, but the acltype is Account or not specified, the network will be automatically assigned to the caller account and domain. " +
```

**Review by weizhouapache — APPROVED** [review]
code lgtm

@DaanHoogland can you review it ? Thanks

**Review by DaanHoogland — APPROVED** [review]

**Comment by DaanHoogland** [comment]
only parameter description fields have changed, merging

**Pull request merged**